target/linux/ramips/dts/TL-WR841NV14.dts

```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include "mt7628an.dtsi"
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+/ {
+	compatible = "tplink,tl-wr841n-v14", "mediatek,mt7628an-soc";
+	model = "TP-Link TL-WR841N v14";
+
+	aliases {
+		led-boot = &led_wlan;
+		led-failsafe = &led_wlan;
+		led-upgrade = &led_wlan;
+	};
+
+	chosen {
+		bootargs = "console=ttyS0,115200";
+	};
+
+	memory@0 {
+		device_type = "memory";
+		reg = <0x0 0x2000000>;
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <20>;
+
+		reset {
+			label = "reset";
+			gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
+			linux,code = <KEY_RESTART>;
+		};
+	};
+
+	leds {
+		compatible = "gpio-leds";
+
+		lan {
+			label = "tl-wr841n-v14:green:lan";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+
+
+		wan_green {
+			label = "tl-wr841n-v14:green:wan";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+		};
+
+		led_wlan: wlan {
+			label = "tl-wr841n-v14:green:wlan";
+			gpios = <&gpio1 9 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&spi0 {
+	status = "okay";
+
+	flash@0 {
+		compatible = "jedec,spi-nor";
+		reg = <0>;
+		spi-max-frequency = <10000000>;
+		m25p,chunked-io = <32>;
+
+		partitions {
+			compatible = "fixed-partitions";
+			#address-cells = <1>;
+			#size-cells = <1>;
+
+			partition@0 {
+				label = "boot";
+				reg = <0x0 0x10000>;
+				read-only;
+			};
+
+			partition@10000 {
+				compatible = "tplink,firmware";
+				label = "firmware";
+				reg = <0x10000 0x3e0000>;
+			};
+
+			factory: partition@3f0000 {
+				label = "factory";
+				reg = <0x3f0000 0x10000>;
+				read-only;
+			};
+		};
+	};
+};
+
+&ehci {
+	status = "disabled";
+};
+
+&ohci {
+	status = "disabled";
+};
+
+&wmac {
+	status = "okay";
+	mtd-mac-address = <&factory 0xf100>;
+	mediatek,mtd-eeprom = <&factory 0x10000>;
+};
+
+&ethernet {
+	mtd-mac-address = <&factory 0xf100>;
+	mediatek,portmap = "wllll";
+};
+
+&pinctrl {
+	state_default: pinctrl0 {
+		gpio {
+			ralink,group = "p4led_an", "p3led_an", "p2led_an", "p1led_an", "p0led_an", "wdt";
```

Comment:
If I counting it right, then `p1led_an` and `p0led_an`  should be probably removed

```
wdt      gpio1 6   reset button
p4led_an gpio1 7   lan led
p3led_an gpio1 8   wan led
p2led_an gpio1 9   wlan led
p1led_an gpio1 10  unused ?
p0led_an gpio1 11  unused ?
```

Reply:
gpio 10 is used by the orange WAN LED that I removed accidentially.

>      * GPIO41, GPIO43: WIFI_led, yellowGreen: 41 low, 43 high
The WIFI led requires the gpio 11 to be set high, maybe this should be explicitly set.
Would this be ok:

`gpio_export {
        compatible = "gpio-export";
        #size-cells = <0>;

        usb_power {
            gpio-export,name = "w";
            gpio-export,output = <1>;
            gpios = <&gpio0 21 GPIO_ACTIVE_HIGH>;
        };
    };`

Reply:
> The WIFI led requires the gpio 11 to be set high

Why is that?

BTW, we're trying to get rid of the `gpio-export` functionality as it's hack for missing kernel functionality, which was rejected in upstream kernel long time ago, for details see this email http://lists.infradead.org/pipermail/openwrt-devel/2019-February/015772.html, discussion in #1366 or https://github.com/openwrt/openwrt/pull/1814#issuecomment-462942022.

So in this case I would probably use gpio-hog, example:

https://github.com/openwrt/openwrt/blob/cf463159df1bee537362d8faee8b1ac51e5bbfef/target/linux/ramips/dts/WN-AC733GR3.dts#L165-L172